osu.Game/Graphics/UserInterface/ProcessingOverlay.cs

```diff
@@ -28,29 +35,54 @@ private void load()
         {
             InternalChildren = new Drawable[]
             {
-                new Box
+                loadingBox = new Container
                 {
-                    Colour = Color4.Black,
-                    RelativeSizeAxes = Axes.Both,
-                    Alpha = 0.9f,
+                    Size = new Vector2(80),
+                    Scale = new Vector2(0.8f),
+                    Masking = true,
+                    CornerRadius = 15,
+                    Anchor = Anchor.Centre,
+                    Origin = Anchor.Centre,
+                    Children = new Drawable[]
+                    {
+                        new Box
+                        {
+                            Colour = Color4.Black,
+                            RelativeSizeAxes = Axes.Both,
+                        },
+                        new LoadingAnimation { State = { Value = Visibility.Visible } }
+                    }
                 },
-                new LoadingAnimation { State = { Value = Visibility.Visible } }
             };
         }
 
-        protected override bool Handle(UIEvent e)
-        {
-            return true;
-        }
+        protected override bool Handle(UIEvent e) => true;
 
         protected override void PopIn()
         {
-            this.FadeIn(transition_duration * 2, Easing.OutQuint);
+            this.FadeIn(transition_duration, Easing.OutQuint);
+            loadingBox.ScaleTo(1, transition_duration, Easing.OutElastic);
+
+            dimTarget?.FadeColour(OsuColour.Gray(0.5f), transition_duration, Easing.OutQuint);
         }
 
         protected override void PopOut()
         {
             this.FadeOut(transition_duration, Easing.OutQuint);
+            loadingBox.ScaleTo(0.8f, transition_duration / 2, Easing.In);
+
+            dimTarget?.FadeColour(Color4.White, transition_duration, Easing.OutQuint);
+        }
+
+        protected override void Dispose(bool isDisposing)
+        {
+            base.Dispose(isDisposing);
+
+            if (State.Value == Visibility.Visible)
+            {
+                // ensure we don't leave the targetin a bad state.
```

Comment:
classic comment nitpick
```suggestion
                // ensure we don't leave the target in a bad state.
```